packages/flutter/lib/src/material/checkbox_theme.dart

```diff
@@ -119,6 +135,8 @@ class CheckboxThemeData with Diagnosticable {
       splashRadius: lerpDouble(a?.splashRadius, b?.splashRadius, t),
       materialTapTargetSize: t < 0.5 ? a?.materialTapTargetSize : b?.materialTapTargetSize,
       visualDensity: t < 0.5 ? a?.visualDensity : b?.visualDensity,
+      shape: ShapeBorder.lerp(a?.shape, b?.shape, t) as OutlinedBorder?,
+      //side: BorderSide.lerp(a?.side, b?.side, t),
```

Comment:
This shouldn't be commented out. Since BorderSide.lerp doesn't support null parameters (for now) you can define a private static utility method, and use ` _lerpSides(a?.side, b?.side, t)`

```dart
  // Special case because BorderSide.lerp() doesn't support null arguments
  static MaterialStateProperty<BorderSide?>? _lerpSides(MaterialStateProperty<BorderSide?>? a, MaterialStateProperty<BorderSide?>? b, double t) {
    if (a == null && b == null)
      return null;
    return _LerpSides(a, b, t);
  }
```